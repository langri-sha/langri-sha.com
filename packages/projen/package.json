{
  "name": "@langri-sha/projen",
  "version": "0.1.0",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*",
    "projen": "0.81.1"
  },
  "peerDependencies": {
    "projen": "^0.81.1",
    "typescript": "^5.0.0"
  },
  "publishConfig": {
    "access": "public"
  }
}
