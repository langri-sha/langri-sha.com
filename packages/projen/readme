# @langri-sha/projen

Collection of [projen] templates for bootstraping monorepos and workspace projects.

[projen]: https://projen.dev/
