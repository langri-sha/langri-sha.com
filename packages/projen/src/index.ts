import {
  Project as BaseProject,
  type ProjectOptions as BaseProjectOptions,
} from 'projen'

export interface ProjectOptions extends BaseProjectOptions {}

export class Project extends BaseProject {
  constructor(options: ProjectOptions) {
    super({
      ...options,
      gitIgnoreOptions: getGitIgnoreOptions(options),
    })
  }
}

const getGitIgnoreOptions = (
  options: ProjectOptions,
): ProjectOptions['gitIgnoreOptions'] => ({
  ...options.gitIgnoreOptions,
  ignorePatterns: [
    ...`
    .*
    !.babelrc
    !.dockerignore
    !.editorconfig
    !.eslintignore
    !.eslintrc.js
    !.gitattributes
    !.gitignore
    !.gitkeep
    !.husky
    !.npmignore
    !.openssl
    !.prettierignore
    !.terraform.lock.hcl
    *.log
    *.tsbuildinfo

    !.github/
    !.husky/
    !.projen/
    dist/
    local/
    node_modules/
    `
      .split('\n')
      .map((pattern) => pattern.trim())
      .filter((pattern) => pattern.length > 0),
    ...(options.gitIgnoreOptions?.ignorePatterns ?? []),
  ],
})
