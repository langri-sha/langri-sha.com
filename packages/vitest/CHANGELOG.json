{
  "name": "@langri-sha/vitest",
  "entries": [
    {
      "date": "Sat, 27 Jul 2024 15:50:17 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d542415a6b2c2c339a8367c77b925842000f9f9a",
            "comment": "feat(workspace): Exclude compiling test modules"
          }
        ]
      }
    },
    {
      "date": "Fri, 26 Jul 2024 21:00:29 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(workspace): Migrate documentation pathnames"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "d58e2b444358fb65166d0878c185220c6e0a7bb0",
            "comment": "feat(projen): Add Git ignore patterns"
          }
        ]
      }
    },
    {
      "date": "Wed, 17 Jul 2024 10:48:15 GMT",
      "version": "0.1.1",
      "tag": "@langri-sha/vitest_v0.1.1",
      "comments": {
        "patch": [
          {
            "author": "email not defined",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "chore(deps): chore(deps): update dependency vitest to v2"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "fix(workspace): Use correct GitHub directory"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "c4b7573f367c165d7c67ed5f259ad53ec9ea5672",
            "comment": "feat(root): Migrate to ESM"
          }
        ]
      }
    },
    {
      "date": "Mon, 08 Jul 2024 21:43:05 GMT",
      "version": "0.1.0",
      "tag": "@langri-sha/vitest_v0.1.0",
      "comments": {
        "minor": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Use `vitest`"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(vitest): Add `nock`"
          }
        ],
        "none": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(workspace): Add repository path"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "feat(tsconfig): Use `${configDir}`"
          }
        ],
        "patch": [
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use correct output directory"
          },
          {
            "author": "filip.dupanovic@gmail.com",
            "package": "@langri-sha/vitest",
            "commit": "94472e1a79e065f4ef9c4e668453e9266b428281",
            "comment": "fix(projen): Use workspace `@langri-sha/tsconfig`"
          }
        ]
      }
    }
  ]
}
