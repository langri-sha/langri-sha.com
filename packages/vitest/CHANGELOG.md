# Change Log - @langri-sha/vitest

This log was last generated on Wed, 17 Jul 2024 10:48:15 GMT and should not be manually modified.

<!-- Start content -->

## 0.1.1

Wed, 17 Jul 2024 10:48:15 GMT

### Patches

- chore(deps): chore(deps): update dependency vitest to v2 (email not defined)
- fix(workspace): Use correct GitHub directory (filip.dupanovic@gmail.com)

## 0.1.0

Mon, 08 Jul 2024 21:43:05 GMT

### Minor changes

- feat(workspace): Use `vitest` (filip.dupanovic@gmail.com)
- feat(vitest): Add `nock` (filip.dupanovic@gmail.com)

### Patches

- fix(projen): Use correct output directory (filip.dupanovic@gmail.com)
- fix(projen): Use workspace `@langri-sha/tsconfig` (filip.dupanovic@gmail.com)
