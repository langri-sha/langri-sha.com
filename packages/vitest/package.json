{
  "name": "@langri-sha/vitest",
  "version": "0.0.0",
  "bugs": {
    "url": "https://github.com/langri-sha/langri-sha.com/issues"
  },
  "repository": {
    "url": "langri-sha/langri-sha.com",
    "path": "/packages/vitest"
  },
  "license": "MIT",
  "author": {
    "name": "Filip DupanoviÄ‡",
    "email": "filip.dupanovic@gmail.com",
    "url": "https://langri-sha.com",
    "organization": false
  },
  "type": "module",
  "main": "src/index.ts",
  "scripts": {
    "prepublishOnly": "rm -rf dist; tsc --project tsconfig.build.json"
  },
  "dependencies": {
    "nock": "13.5.4",
    "tempy": "1.0.1"
  },
  "devDependencies": {
    "@langri-sha/tsconfig": "workspace:*"
  },
  "peerDependencies": {
    "vitest": "^1.6.0"
  },
  "publishConfig": {
    "access": "public",
    "main": "dist/index.js",
    "types": "dist/index.d.ts"
  },
  "//": "~~ Generated by projen. To modify, edit .projenrc.mts and run \"npx projen\"."
}
