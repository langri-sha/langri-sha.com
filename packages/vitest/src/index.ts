export * from 'vitest'
export { default as nock } from 'nock'
export { default as tempy } from 'tempy'
