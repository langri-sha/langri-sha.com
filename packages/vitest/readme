# @langri-sha/vitest

Provides some useful helpers that are commonly used for authoring tests.

## Usage

Install the required dependencies:

```sh
npm install -D vitest @langri-sha/vitest
```

Then import your Vitest dependencies from here:

```js
// some.test.js
import { expect, test, tempy } from '@langri-sha/vitest'

test(/*...*/)
```

## See

- [`@jest/globals`]
- [`tempy`]

[`@jest/globals`]: https://jestjs.io/docs/api
[`tempy`]: https://github.com/sindresorhus/tempy
