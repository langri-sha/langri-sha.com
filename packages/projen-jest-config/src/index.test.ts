import { synthSnapshot } from 'projen/lib/util/synth'
import { expect, test } from '@langri-sha/vitest'

import { Project } from 'projen'
import { JestConfig } from './index'

test('defaults', () => {
  const project = new Project({
    name: 'test-project',
  })

  new JestConfig(project)

  project.synth()
  expect(synthSnapshot(project)['jest.config.ts']).toMatchSnapshot()
})

test('with filename', () => {
  const project = new Project({
    name: 'test-project',
  })

  new JestConfig(project, { filename: 'jest.config.mts' })

  project.synth()
  expect(synthSnapshot(project)['jest.config.mts']).toMatchSnapshot()
})

test('with extends', () => {
  const project = new Project({
    name: 'test-project',
  })

  new JestConfig(project, { extends: '@langri-sha/jest-config' })

  project.synth()
  expect(synthSnapshot(project)['jest.config.ts']).toMatchSnapshot()
})

test('with config', () => {
  const project = new Project({
    name: 'test-project',
  })

  new JestConfig(project, { config: { testEnvironment: 'node' } })

  project.synth()
  expect(synthSnapshot(project)['jest.config.ts']).toMatchSnapshot()
})
