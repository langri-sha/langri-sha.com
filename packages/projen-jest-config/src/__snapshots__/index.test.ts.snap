// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults 1`] = `
"import type { Config } from 'jest'

    const config: Config = {
      
      }

    export default config
    "
`;

exports[`with config 1`] = `
"import type { Config } from 'jest'

    const config: Config = {
      
      "testEnvironment":"node"}

    export default config
    "
`;

exports[`with extends 1`] = `
"import type { Config } from 'jest'
import defaults from '@langri-sha/jest-config'

    const config: Config = {
      ...defaults,
      }

    export default config
    "
`;

exports[`with filename 1`] = `
"import type { Config } from 'jest'

    const config: Config = {
      
      }

    export default config
    "
`;
